### DISCUSSION TITLE
Override Fragment Shader in WebGLTile (Layer)?

### DISCUSSION THREAD

**Discussion by jarzynka** [openlayers/openlayers#16217] [Q&A]
I've been working at this problem for over a week and hope I've done my due diligence at research before posting here.  I even tried several GPT rounds to no success.  I've been using Open Layers for 4 months and love it!  I am consuming two sets of raster tiles (XYZ) - one of a globe map and the second one a map without the land - just the water features.  I'd like to play around with GL water effects but am just trying to change the texture color (which is blue) to red.  

I know I can do this with the style.color field but I want to see if I can override the fragmentShader property to start working on the water element.  No matter what I do, the texture from the XYZ server remains - even if I delete the ol/layers/WebGLTile.js/ts file(s). 

I'm not sure if the fragment shader isn't compiling properly or if I'm completely trying to override it incorrectly.  Does anyone have any suggestions?  Do I have to subclass WebGLTile.  I see there are properties for vertexShader and fragmentShader but they appear to have no effect and log nothing to the console.  Thanks!

**Comment by kingkong-cmd** [comment]
Just my 5 cents. I did some custom GL stuff here https://github.com/openlayers/openlayers/discussions/16196. Understanding how to access and interact with the GL under the hood of OL was so incomprehensible that it was easier to just subclass a Layer and build the GL from scratch, drawing directly on the canvas of the layer. On the plus side, this enabled me to use twgl.js as an abstraction layer for the GL, which helped quite a lot, but comes with many negatives as well.

It would be awesome if OL also had a more understandable API for working with GL, but as far as I managed to understand, this is only for those with deep knowledge of the inner workings of OL.
